### DISCUSSION TITLE
OnlyShowIn= entry not being honored in .desktop files

### DISCUSSION THREAD

**Discussion by edwardp3** [lxqt/lxqt#2396] [Q&A] [answered]
I'm trying to clean up the desktops. I have two .desktop files for Home, one for PCManFM, the other for PCManFM-Qt. Both LXDE and LXQt are installed (Debian sid).

```
OnlyShowIn=LXDE;
OnlyShowIn=LXQt;
```
The PCManFM .desktop file has the above LXDE line in it, the Qt .desktop file has the other, yet both icons appear on both desktops, after logging out and also after reboots.  The two files have different .desktop filenames.

Is there a reason why _OnlyShowIn_ isn't being honored? It is part of the [FreeDesktop specification](https://specifications.freedesktop.org/menu-spec/latest/ar01s03.html) and they are correctly ending with a semi-colon. 

Both desktops are listed at [Registered OnlyShowIn Environments](https://specifications.freedesktop.org/menu-spec/latest/apb.html).

![home-icons-lxde](https://user-images.githubusercontent.com/73129450/227027927-35ed76f5-ae2b-4f37-a982-ea3afe51f44a.png)
![home-icons-lxqt](https://user-images.githubusercontent.com/73129450/227027928-34fcee52-c486-4f31-a20f-336d0d489fff.png)

Thanks in advance.

**Comment by stefonarch** [comment, marked as answer]
I saw that `OnlyShowIn=LXQt;` is well honored in the quicklaunch plugin (besides in the main menu plugin which is the goal). 

I don't think the desktop  does honor this display setting - basically it's a folder and you can see also all `.desktop` files  in `/usr/share/applications`.

**Reply by edwardp3** [reply]
Both of the .desktop files are in ~/Desktop

I copied the entry for PCManFM-Qt from _/usr/share/applications_ into ~/Desktop and only added the _OnlyShowIn_ line to it. 

For the LXQt entry below, I removed all of the foreign (non-English) language lines, before copying/pasting here. The original file retains the removed text, except for the _OnlyShowIn_ addition.

```
[Desktop Entry]
Type=Application
Exec=pcmanfm-qt %U
MimeType=inode/directory;
Icon=/usr/share/icons/nuoveXT2/128x128/places/user-home.png
Categories=FileManager;Utility;Core;Qt;
StartupNotify=true
Name=Home Qt
OnlyShowIn=LXQt;
GenericName=File Manager
Comment=Browse the file system and manage the files

```

PCManFM (LXDE):

```
[Desktop Entry]
Type=Application
Exec=pcmanfm
Icon=/usr/share/icons/nuoveXT2/128x128/places/user-home.png
OnlyShowIn=LXDE;
Terminal=false
StartupNotify=true
MimeType=x-directory/normal;inode/directory;
Name=Home
```

**Reply by tsujan** [reply]
As [Desktop Entry Specification](https://specifications.freedesktop.org/desktop-entry-spec/desktop-entry-spec-1.1.html) says, a desktop entry is about "how a particular program is to be launched, how it appears in menus, etc." Inside folders, a desktop entry is just a file, which can be executed under certain circumstances and in certain ways.

> I copied the entry for PCManFM-Qt from /usr/share/applications into ~/Desktop 

`~/Desktop` is a folder that's drawn on the background. `OnlyShowIn` is not relevant inside folders; it's for menus or anywhere a menu hierarchy is displayed.

**Reply by edwardp3** [reply]
Thanks.

**Reply by edwardp3** [reply]
I installed a third desktop, Xfce. With the `OnlyShowIn` line removed from both .desktop files, the icons appear on the Xfce desktop. 

With the lines added back in, they do **not** appear on the desktop, which is the desired result with the LX* desktops.

**Reply by stefonarch** [reply]
So basically yours would be a feature request?

Personally I think the desktop is not the place from which to start applications. There is the menu, lxqt-runner, shortcuts and quickstart plugin in one or more panels, hiding maybe.

**Reply by tsujan** [reply]
> .... which is the desired result with the LX* desktops.

Treating a Desktop file as hidden just because of `OnlyShowIn` is inconsistent with other folders; hiding such files in all folders is worse — to say nothing of the fact that pcmanfm-qt's desktop module can be used outside LXQt.

> I installed a third desktop, Xfce.

If you installed GNOME, KDE, Enlightenment,.., you would see different things: no desktop item at all, "Plasma widgets", items+ widgets,.... Each DE has its own path.

**Reply by edwardp3** [reply]
OK, thank you.
